packages/block-library/src/image/interactivity.js

```diff
@@ -0,0 +1,138 @@
+					context.core.image.lastFocusedElement =
+						window.document.activeElement;
+					context.core.image.scrollPosition = window.scrollY;
+
+					context.core.image.documentElement =
+						document.documentElement;
+					context.core.image.adminElement =
+						document.querySelector( '#wpadminbar' );
+					context.core.image.siteBlocksElement =
+						document.querySelector( '.wp-site-blocks' );
+
+					context.core.image.documentElement.classList.add(
+						'has-lightbox-open'
+					);
+
+					if ( context.core.image.adminElement ) {
+						context.core.image.adminElement.setAttribute(
+							'inert',
+							true
+						);
+					}
+
+					if ( context.core.image.siteBlocksElement ) {
+						context.core.image.siteBlocksElement.setAttribute(
+							'inert',
+							true
+						);
+					}
+				},
+				hideLightbox: async ( { context, event } ) => {
+					if ( context.core.image.lightboxEnabled ) {
+						// If scrolling, wait a moment before closing the lightbox.
+						if (
+							event.type === 'mousewheel' &&
+							Math.abs(
+								window.scrollY -
+									context.core.image.scrollPosition
+							) < 5
+						) {
+							return;
+						}
+						document.documentElement.classList.remove(
+							'has-lightbox-open'
+						);
+
+						if ( context.core.image.adminElement ) {
+							context.core.image.adminElement.removeAttribute(
+								'inert'
+							);
+						}
+
+						if ( context.core.image.siteBlocksElement ) {
+							context.core.image.siteBlocksElement.removeAttribute(
+								'inert'
+							);
+						}
+
+						context.core.image.lightboxEnabled = false;
+						context.core.image.lastFocusedElement.focus();
+					}
+				},
+				handleKeydown: ( { context, actions, event } ) => {
+					if ( context.core.image.lightboxEnabled ) {
+						if ( event.key === 'Tab' || event.keyCode === 9 ) {
+							// If shift + tab it change the direction
+							if (
+								event.shiftKey &&
+								window.document.activeElement ===
+									context.core.image.firstFocusableElement
+							) {
+								event.preventDefault();
+								context.core.image.lastFocusableElement.focus();
+							} else if (
+								! event.shiftKey &&
+								window.document.activeElement ===
+									context.core.image.lastFocusableElement
+							) {
+								event.preventDefault();
+								context.core.image.firstFocusableElement.focus();
+							}
+						}
+
+						if ( event.key === 'Escape' || event.keyCode === 27 ) {
+							actions.core.image.hideLightbox( {
+								context,
+								event,
+							} );
+						}
+					}
+				},
+			},
+		},
+	},
+	effects: {
+		core: {
+			image: {
+				initLightbox: async ( { context, ref } ) => {
+					const focusableElements =
+						ref.querySelectorAll( focusableSelectors );
+					context.core.image.firstFocusableElement =
+						focusableElements[ 0 ];
+					context.core.image.lastFocusableElement =
+						focusableElements[ focusableElements.length - 1 ];
```

Comment:
Should we move this inside the `if ( context.core.image.lightboxEnabled )` conditional? We only want to initialize them when the lightbox is opened, right?

Reply:
Sure thing, this has been changed 👍